docs/config.html

```diff
@@ -484,6 +485,65 @@ <h2 id="build-env" class="title-2">[BuildEnv]</h2>
   </p>
 </section>
 
+<section class="mt4">
+  <h2 id="sandbox" class="title-2">[Sandbox]</h2>
+  <ul class="bulleted-list">
+    <li>
+      <div>
+        <h3 class="mt1 f6 lh-title">
+          Tool
+        </h3>
+
+        <p>
+          The binary to use for sandboxing build/test actions. Typically this
+          is <code>please_sandbox</code>.<br />
+          It receives the program to execute and all its arguments in its <code>argv</code>,
+          it is expected to do whatever it needs to then <code>exec</code> the real action.
+        </p>
+      </div>
+    </li>
+    <li>
+      <div>
+        <h3 class="mt1 f6 lh-title">
+          Build <span class="normal">(bool)</span>
+        </h3>
+
+        <p>
+          Activates sandboxing for individual build actions, which isolates them
+          from network access, IPC and the repo-local filesystem. Currently
+          only works on Linux. Defaults to <code class="code">False</code>.
+        </p>
+      </div>
+    </li>
+    <li>
+      <div>
+        <h3 class="mt1 f6 lh-title">
+          Test <span class="normal">(bool)</span>
+        </h3>
+
+        <p>
+          Activates sandboxing for individual test actions, which isolates them
+          from network access, IPC and the repo-local filesystem. Currently
+          only works on Linux. Defaults to <code class="code">False</code>.
+        </p>
+      </div>
+    </li>
+    <li>
+      <div>
+        <h3 class="mt1 f6 lh-title">
+          Dir <span class="normal">(repeated string)</span>
+        </h3>
+
+        <p>
+          Directories that the sandbox tool should mask (it's unspecified how, although mounting an
+          empty filesystem over the top is the current implementation).<br />
+          These are passed to the tool itself in the <code>SANDBOX_DIRS</code> env var as a comma-separated list.
```

Comment:
Should we not whitelist directories? We can probably determine this from the PATH, SECRETS and TOOLS variables the rule requires. Maybe that can come in a future release.

Reply:
Those variables aren't sufficient to determine the set of directories you need - for example you might have `/usr/bin` on the PATH, but what do you need for things in there to run? Probably `/usr`, `/lib`, `/lib64`, `/bin` but you might also need some subsets of `/etc` or `/var` depending on what they're doing.

We can definitely think about extending this in future, but this ability to blacklist directories has definitely been useful internally.